### ISSUE TITLE
Rerouting in parallel execution with CPP planner not possible

### ISSUE THREAD

**Issue by alessiogambi** [TUM-AVS/Frenetix-Motion-Planner#2]
Hi all, 

# The problem

I am trying to dynamically change the route of a CPP Frenetix Planner (executed in parallel mode) after the planner has been started. However, calling `set_reference_and_coordinate_system` with a newly computed reference path hangs (without any error message) while instantiating a new `CoordinateSystem` (from the `utils_coordinate_systems` module). I suspect that instantiating such class in a subprocess is (for whatever reason) impossible or causes some deadlock/race condition. 

> Note: Also, the logger does not work because it tries to insert in the reference path twice in the DB with the same key

For reference, I am running the code:
- using Python 3.10
- Ubuntu WSL (Windows 11)
- without CUDA
- frentix 0.3.1

# What I am trying to achieve?

The Frenetix planner in some cases cannot find a physically feasible solution, so I want to recompute another solution (call it an Emergency solution) such that:
- the planner does not care about road and other vehicles/obstacles (it is ok if the planner goes out of the road or collides with other objects)
- the reference path is a straight line (from the planner center following its orientation)
- the optimal trajectory is the one that takes the vehicle to a stopping position as soon as possible. (I have a special cost function for that)

Basically, a kind of AEB system. I have already implemented the code that computes all of the above but requires updating the coordinate system of the planner such that I can simply reinvoke the "plan()" method. I would like to avoid recomputing the trajectory manually (since there's the planner for that)

Any help is appreciated. Let me know what other information you'd like to have
